addons/mail_bot/models/discuss_channel.py

```diff
@@ -10,3 +9,7 @@ class Channel(models.Model):
     def execute_command_help(self, **kwargs):
         super().execute_command_help(**kwargs)
         self.env['mail.bot']._apply_logic(self, kwargs, command="help")  # kwargs are not usefull but...
+
+    def _message_post_after_hook(self, message, msg_vals):
+        self.env["mail.bot"]._apply_logic(self, msg_vals)
```

Comment:
If you move it here, it should be removed from thread.

It looks like the 2 removed files were not properly backported.

Reply:
Bad picking on my part :)